src/mame/trs/agvision.cpp

```diff
@@ -0,0 +1,417 @@
+}
+
+//-------------------------------------------------
+//  ff20_write
+//-------------------------------------------------
+
+void agvision_state::ff20_write(offs_t offset, uint8_t data)
+{
+	m_rs232->write_txd(data & 0x80 ? 1 : 0);
+ 	m_vdg->gm0_w(data & 0x08 ? ASSERT_LINE : CLEAR_LINE);
+	m_vdg->gm1_w(data & 0x04 ? ASSERT_LINE : CLEAR_LINE);
+	m_vdg->gm2_w(data & 0x02 ? ASSERT_LINE : CLEAR_LINE);
+	m_vdg->ag_w(data & 0x01 ? ASSERT_LINE : CLEAR_LINE);
+}
+
+//-------------------------------------------------
+//  configure_sam
+//-------------------------------------------------
+
+void agvision_state::configure_sam()
+{
+	offs_t ramsize = m_ram->size();
+	m_sam->space(0).install_ram(0, ramsize - 1, m_ram->pointer());
+	if (ramsize < 65536)
+		m_sam->space(0).nop_readwrite(ramsize, 0xffff);
+}
+
+//-------------------------------------------------
+//  timer_elapsed
+//-------------------------------------------------
+
+TIMER_CALLBACK_MEMBER(agvision_state::timer_elapsed)
+{
+	m_timer->adjust(attotime::from_usec(CD_DELAY));
+	m_pia_0->cb1_w(m_cd);
+	m_cd = !m_cd;
+}
+
+//-------------------------------------------------
+//  pia0_pa_r - keyboard handler
+//-------------------------------------------------
+
+uint8_t agvision_state::pia0_pa_r()
+{
+	uint8_t pia0_pb = m_pia_0->b_output();
+
+	uint8_t pia0_pa = 0x7F;
+
+	/* poll the keyboard, and update PA6-PA0 accordingly*/
+	for (unsigned i = 0; i < m_keyboard.size(); i++)
+	{
+		int value = m_keyboard[i]->read();
+		if ((value | pia0_pb) != 0xFF)
+		{
+			pia0_pa &= ~(0x01 << i);
+		}
+	}
+
+	pia0_pa |= (m_rs232->rxd_r() ? 0x80 : 0);
+
+	return pia0_pa;
+}
+
+//-------------------------------------------------
+//  hang_up
+//-------------------------------------------------
+
+INPUT_CHANGED_MEMBER(agvision_state::hang_up)
+{
+ 	m_timer->adjust(attotime::never);
+}
+
+//-------------------------------------------------
+//  pia0_cb2_w
+//-------------------------------------------------
+
+void agvision_state::pia0_cb2_w(int state)
+{
+	if( state == 1 )
+	{
+		m_timer->adjust(attotime::from_usec(CD_DELAY*8000));
+	}
+	else
+	{
+		m_timer->adjust(attotime::never);
+	}
+}
+
+//**************************************************************************
+//  ROMS
+//**************************************************************************
+
+ROM_START(agvision)
+	ROM_REGION(0xa000,MAINCPU_TAG,0)
+	ROM_LOAD("8041716-1.1-agvision.u13", 0x0000, 0x0800, CRC(5b11b13e) SHA1(896e68f90ed57e0921887e717fc92eda067a210d))
+ROM_END
+
+ROM_START(trsvidtx)
+	ROM_REGION(0xa000,MAINCPU_TAG,0)
+	ROM_LOAD("8041716-1.1-videotex.u13", 0x0000, 0x0800, CRC(821a59bb) SHA1(e3643f27fcf8287c0bc0f66b21554dc988ded9c1))
+ROM_END
+
+//    YEAR  NAME      PARENT COMPAT MACHINE   INPUT     CLASS           INIT        COMPANY              FULLNAME    FLAGS
+COMP( 1979, agvision, 0,     0,     agvision, agvision, agvision_state, empty_init, "Tandy Radio Shack", "AgVision", MACHINE_SUPPORTS_SAVE | MACHINE_NO_SOUND_HW )
```

Comment:
Should company be or at least mention Elanco since the machine is branded that way?

Reply:
I have this question out to our community historian.

Reply:
We discussed this and determined that yes, the company should be Elanco.